api/config/v1alpha1/envoygateway_types.go

```diff
@@ -120,6 +120,19 @@ type Gateway struct {
 	//
 	// +optional
 	ControllerName string `json:"controllerName,omitempty"`
+
+	// ExtensionAPIs defines the settings related to specific Gateway API Extensions
```

Comment:
Is this comment correct?

Reply:
any suggestions for improving it ?

Reply:
The comment (and the name ExtensionAPIs as well) looks vague to me. Did you intentionally make it this way because this API will also serve other purposes other just than disabling/enabling specific features?

Reply:
yeah sorta, trying to use a high level noun that can encapsulate an entire class of settings, so chose `ExtensionAPIs` to signify all settings pertaining to it

Reply:
EnvoyPatchPolicy is not part of the Gateway API. That's why I feel confused when I saw the location of the ExtensionAPIs(it's part of the Gateway Structure) and the comment “ ExtensionAPIs defines the settings related to specific Gateway API Extensions implemented by Envoy Gateway”.

Reply:
any suggestions on what might be a better home ?

Reply:
For any features we're consider dangerous of not stable, we can pass them as command line parameters like [K8s feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) does.

--feature-gates=...,EnvoyPatchPolicy=true

Reply:
@zhaohuabing yes, that's what I'm thinking.

On the other hand, EnvoyPatch will before there's a CR applied, so maybe it's unneccessary to do this for EnvoyPatch?

Reply:
based on this discussion I believe there are 3 approaches
1. add a `--feature-flags` flag to the EG CLI
2. add a top level `FeatureFlags` struct to the `EnvoyGateway` config
```
featureFlags:
  enableEnvoyPatchPolicy: true
```  
3. Continue with the above approach
```
gateway:
  extenstionAPIs:
    enableEnvoyPatchPolicy: true
```

I'm a -1 on 1. since I'd prefer if we kept all our config knobs inside the `EnvoyGateway` config instead of creating  many CLI flags and its easier to manage doc generation with the `EnvoyGateway` config.

Fine with 2. or 3. , 2. is a flat approach and might end up having many unrelated knobs living next to each other, but that might be tolerable.

@zhaohuabing @zirain @AliceProxy any thoughts ?

Reply:
I agree with your reasons on 3, but still the location and the name of the structure `extenstionAPIs ` seem not quite aligned with its actual function. But I can't figure out a better name and location either.

Reply:
Maybe we can hold this until we have other settings that we want to put in the `ExtensionAPIs`?  By that time, we can revisit this matter and potentially come up with a better way to handle it.

Reply:
we will need to fix this before v0.5.0, since we want `EnvoyPatchPolicy` to be opt in

Reply:
@kflynn @AliceProxy and I discussed this issue in the community meeting today, and favor 3.
Although `ExtentionAPIs` is not the best name, we cannot come up with a better name at this point so prefer sticking to it for now,

Reply:
> we will need to fix this before v0.5.0, since we want `EnvoyPatchPolicy` to be opt in

@arkodg @AliceProxy is this a must-have in v0.5.0? As @zirain said, if no EnvoyPatchPolicies are created, it has no impact at all.

Reply:
chatted with @zhaohuabing a little more, and he shared his hesitation with
```
gateway:
  extensionAPIs:
    ...
```
where `extensionAPIs` are not part of the Gateway API, which is valid.

another option is to reuse the top level `extension` field, that is used by the extension service
```
extension:
  apis:
    enableEnvoyPatchPolicy: true
  controlPlane:
    hooks:
    service:
```

Reply:
Gateway can be the container of the settings of the whole EG, not just Gateway API. 

I suggest a minor change to the comment to address this:

Change 

"Gateway defines the desired Gateway API configuration of Envoy Gateway."

To

"Gateway defines the desired API configuration of Envoy Gateway."